### PR TITLE
Add function for custom selection for columns of required data. Brilliant!

### PR DESCRIPTION
**Pull request by zztin**
1. Organize go() function in generic.py to several compartments. 
2. Add a new function to produce flexibility to select required columns in data to be analysed.
3. Currently, the user input has to be in the correct format (i.e. integer), if not, the program will crash. 
4. in atta_cli.py change default filename "readcsv.csv" to "data.csv" to indicate this is the filename for data.

### CODE REVIEW

**Comment by tai271828** [comment]
Hi, thanks for the PR. Few suggestions:

1. Please add description of this PR is possible. It would be much better it also includes how we could test this PR.
2. The jupyter notebook looks like an accident commit?
3. Please add "new line" (see inline comment)

File: atta/controller/atta_cli.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import click
 
 @click.command()
-@click.option('--readcsv', default="readcsv.csv", help='read csv format data')
+@click.option('--readcsv', default="data.csv", help='read csv format data')
```
Comment (tai271828) [inline atta/controller/atta_cli.py]: Why do you change this name?

File: atta/analyzer/generic.py [hunk 2]
```diff
@@ -8,199 +8,65 @@
...
+    logging.debug(df.keys())
+    # select necessary columns from df
+    df = select_column(df)
+    df = add_cat_title(df)
+    plot_counts(df,year)
+    print('Go process finished completely. Find results in the same directory as the csv file.')
 
 def fake_go():
-    print('fake_go is executed.')
+    print('fake_go is executed.')
```
Comment (tai271828) [inline atta/analyzer/generic.py]: A common recommended coding style is to add an new line at the end of the file. Please add it. Thanks.

File: atta/controller/atta_cli.py (hunk 1 above)
Comment (zztin) [reply atta/controller/atta_cli.py]: I thought with the name "data.csv" is more clear to indicate this is just the data file.

Comment (tai271828) [reply atta/controller/atta_cli.py]: This is good idea. However the other part need to be changed in the meantime. Like def main(readcsv). Would you please follow README and make sure `atta --data xxx.csv` still work after you change the click.option and its associated code? Then we could call "the new algorithm is integrated into the original comment line tool".

Comment (zztin) [reply atta/controller/atta_cli.py]: I think you misunderstand the change. This only change the default file name, not the command line command. Therefore it still work as atta --readcsv xxx.csv
I have tested it and it worked.
Nevertheless, if you like to keep the original default filename as readcsv.csv, it is also fine with me.

Comment (tai271828) [reply atta/controller/atta_cli.py]: Right. Thanks for the elaboration. \o/

**Comment by tai271828** [comment]
Much better! Thanks for your contribution.

**Pull request merged**